netmaster/master/network.go

```diff
@@ -162,34 +163,62 @@ func CreateNetwork(network intent.ConfigNetwork, stateDriver core.StateDriver, t
 }
 
 func attachServiceContainer(tenantName, networkName string, stateDriver core.StateDriver) error {
-	contName := tenantName + "dns"
+	contName := getDNSName(tenantName)
 	docker, err := utils.GetDockerClient()
 	if err != nil {
 		log.Errorf("Unable to connect to docker. Error %v", err)
 		return err
 	}
 
+	cinfo, err := docker.InspectContainer(contName)
+	if err != nil {
+		if strings.Contains(err.Error(), "no such id") {
+			// DNS container not started for this tenant. Start skydns container
+			err = startServiceContainer(tenantName)
+			if err != nil {
+				log.Warnf("Error starting service container. "+
+					"Continuing without DNS provider. Error: %v", err)
+				return nil
+			}
+			cinfo, err = docker.InspectContainer(contName)
+			if err != nil {
```

Comment:
@DivyaVavili the change looks good. we might want to "return nil" in case this inspect gives an error. Not sure If we can depend on the cinfo value in case of errors.